rust/toolchain.bzl

```diff
@@ -197,6 +207,15 @@ def _out_dir_setup_cmd(out_dir_tar):
 # The rust_toolchain rule definition and implementation.
 
 def _rust_toolchain_impl(ctx):
+  compilation_mode_opts = {}
+  for k, v in ctx.attr.opt_level.items():
```

Comment:
Optional Nit:

I think I personally prefer just using the known variants of COMPILATION_MODE, rather than letting this be freeform. That is to say, just iterate over ["opt", "dbg", "fastbuild"], and disregard other values.

That is to say, I think that the toolchain parameters probably have too much flexibility.